package/MDAnalysis/core/topologyattrs.py

```diff
@@ -59,6 +59,34 @@
 from .. import _TOPOLOGY_ATTRS, _TOPOLOGY_TRANSPLANTS, _TOPOLOGY_ATTRNAMES
 
 
+def _check_dtype(func):
+    def _attr_dtype(values):
+        # check dtype of values array
+        dtypes = ['int8','int16','int32','int64']
```

Comment:
This is a correct check for the attrs which are strictly ints (eg atom ids, resids).  Some attrs are strings (which we store as numpy arrays of `object` dtype.  I think with this approach we'll need a decorator for each case

Reply:
So should i increase the extend the datatypes in the dtypes array